lib/types/cql-utils.js

```diff
@@ -81,23 +81,37 @@ function encodeMap(value, parentType) {
         }
         const val = value[key];
         res.push([
-            getWrapped(keySubtype || guessTypeChecked(key), key),
-            getWrapped(valueSubtype || guessTypeChecked(val), val),
+            getWrappedType(keySubtype || guessTypeChecked(key), key),
+            getWrappedType(valueSubtype || guessTypeChecked(val), val),
         ]);
     }
     return res;
 }
 
 /**
- * Wrap value into QueryParameterWrapper based on the type
+ * Wrap value into MaybeUnsetQueryParameterWrapper based on the type
  * @param {rust.ComplexType} type
  * @param {*} value
- * @returns {rust.QueryParameterWrapper}
+ * @returns {rust.MaybeUnsetQueryParameterWrapper}
  */
 function getWrapped(type, value) {
-    if (value === null || value === types.unset) {
+    if (value === null) {
         return null;
+    } else if (value === types.unset) {
+        return rust.MaybeUnsetQueryParameterWrapper.unset();
     }
+    return rust.MaybeUnsetQueryParameterWrapper.fromType(
+        getWrappedType(type, value),
+    );
```

Comment:
:thinking: Here, we have a mess about naming. `QueryParameterWrapper` is **not** a type; it's a value. Of course, the value has some type, too, but it's more than just a type. `CqlType` is a type; `CqlValue` is a value.

As the naming you chose has brought me the same confusion many times, I believe it will be the same for other future readers. Let's protect them.

```suggestion
    return rust.MaybeUnsetQueryParameterWrapper.fromNonNullNonUnsetValue(
        wrapValueBasedOnType(type, value),
    );
```